main/templates/main/index.html

```diff
@@ -4,6 +4,15 @@
 <h3> These are the matches</h3>
 {% endblock %}
 {% block content %}
+<form action="{% url 'api_matches'%}" method="POST">
+    {% csrf_token %}
+    <input type="json" name="player1" id="player1" placeholder="player1 id" />
```

Comment:
I don't think there's a `json` input type. See [the docs](https://www.w3schools.com/tags/att_input_type.asp) for possible values.
I believe the player names should be text, the scores should be number (you can define min and max on them in html), and all the id fields should be `select` fields instead of `input`. The options have to be passed down from the view to the template.